File: app/src/test/java/com/myapplication/rick/packagedelivery/TestRangeClass.java
package com.myapplication.rick.packagedelivery;

import org.junit.Test;
import static org.junit.Assert.*;

/**
 * To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
public class TestRangeClass {
    private final int lowerBoundUneven;
    private final int upperBoundUneven;
    private final int lowerBoundEven;
    private final int upperBoundEven;
    private final int zeroLowerBound;
    private final int zeroUpperBound;

    public TestRangeClass() {
        lowerBoundUneven = 1;
        upperBoundUneven = 89;
        lowerBoundEven = 2;
        upperBoundEven = 90;
        zeroLowerBound = 0;
        zeroUpperBound = 0;
    }

    @Test
    public void rangeCreationUnevenValid() {
        Range range = new Range(lowerBoundUneven, upperBoundUneven, RangeType.Uneven);
    }

    @Test
    public void rangeCreationEvenValid() {
        Range range = new Range(lowerBoundEven, upperBoundEven, RangeType.Even);
    }
    @Test
    public void rangeCreationAllNumbersValid() {
        Range range = new Range(lowerBoundEven, upperBoundUneven, RangeType.All);
    }

    @Test
    public void getMethodsAndAllParametersSetRight() {
        Range range = new Range(lowerBoundEven, upperBoundUneven, RangeType.All);
        assertEquals(range.getLowerBound(), lowerBoundEven);
        assertEquals(range.getUpperBound(), upperBoundUneven);
        assertEquals(range.getRangeType(), RangeType.All);
    }

    @Test (expected = IllegalArgumentException.class)
    public void rangeCreationInvalidLowerBound() throws IllegalArgumentException {
        Range range = new Range(zeroLowerBound, upperBoundUneven, RangeType.All);
    }

    @Test (expected = IllegalArgumentException.class)
    public void rangeCreationInvalidUpperBound() throws IllegalArgumentException {
        Range range = new Range(lowerBoundEven, zeroUpperBound, RangeType.All);
    }
    @Test (expected = IllegalArgumentException.class)
    public void rangeCreationLowerBoundGreaterThanUpperBound() throws IllegalArgumentException {
        Range range = new Range(upperBoundEven, lowerBoundEven, RangeType.All);
    }
    @Test (expected = IllegalArgumentException.class)
    public void rangeCreationLowerBoundEvenWithRangeTypeUneven() throws IllegalArgumentException {
        Range range = new Range(lowerBoundEven, upperBoundUneven, RangeType.Uneven);
    }
    @Test (expected = IllegalArgumentException.class)
    public void rangeCreationLowerBoundUnevenWithRangeTypeEven() throws IllegalArgumentException {
        Range range = new Range(lowerBoundUneven, upperBoundEven, RangeType.Even);
    }
    @Test (expected = IllegalArgumentException.class)
    public void rangeCreationWithNullRangeType() throws IllegalArgumentException {
        Range range = new Range(lowerBoundEven, upperBoundEven, null);
    }
}

File: app/src/test/java/com/myapplication/rick/packagedelivery/TestRangeClass.kt
package com.myapplication.rick.packagedelivery

import org.junit.Test
import org.junit.Assert.*

/**
 * To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
class TestRangeClass {
    private val lowerBoundUneven: Int = 1
    private val upperBoundUneven: Int = 89
    private val lowerBoundEven: Int = 2
    private val upperBoundEven: Int = 90
    private val zeroLowerBound: Int = 0
    private val zeroUpperBound: Int = 0

    @Test
    fun rangeCreationUnevenValid() {
        val range = Range(lowerBoundUneven, upperBoundUneven, RangeType.Uneven)
    }

    @Test
    fun rangeCreationEvenValid() {
        val range = Range(lowerBoundEven, upperBoundEven, RangeType.Even)
    }

    @Test
    fun rangeCreationAllNumbersValid() {
        val range = Range(lowerBoundEven, upperBoundUneven, RangeType.All)
    }

    @Test
    fun getMethodsAndAllParametersSetRight() {
        val range = Range(lowerBoundEven, upperBoundUneven, RangeType.All)
        assertEquals(range.lowerBound.toLong(), lowerBoundEven.toLong())
        assertEquals(range.upperBound.toLong(), upperBoundUneven.toLong())
        assertEquals(range.rangeType, RangeType.All)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun rangeCreationInvalidLowerBound() {
        val range = Range(zeroLowerBound, upperBoundUneven, RangeType.All)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun rangeCreationInvalidUpperBound() {
        val range = Range(lowerBoundEven, zeroUpperBound, RangeType.All)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun rangeCreationLowerBoundGreaterThanUpperBound() {
        val range = Range(upperBoundEven, lowerBoundEven, RangeType.All)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun rangeCreationLowerBoundEvenWithRangeTypeUneven() {
        val range = Range(lowerBoundEven, upperBoundUneven, RangeType.Uneven)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun rangeCreationLowerBoundUnevenWithRangeTypeEven() {
        val range = Range(lowerBoundUneven, upperBoundEven, RangeType.Even)
    }

    @Test
    fun rangeCreationOneNumber() {
        val range = Range(lowerBoundEven, lowerBoundEven, RangeType.All)
    }
}